Src/DependencyCollector/Shared/Implementation/HttpDiagnosticSourceListener.cs

```diff
@@ -100,6 +105,81 @@ protected virtual void Dispose(bool disposing)
                 this.disposed = true;
             }
         }
+
```

Comment:
Reflection is slow, why not reference the properties directly through their getters?